app/javascript/utils/typecheck.ts

```diff
@@ -0,0 +1,14 @@
+export function typecheck<T>(json: any, key: string): T | never {
```

Comment:
```suggestion
/**
 * Checks if the json has a top-level key. If it does, types the value at
 * that top-level key as T.
 */
export function typecheck<T>(json: unknown, key: string): T | never {
```

Probably more accurate.